crates/rpc-types-engine/src/sidecar.rs

```diff
@@ -0,0 +1,102 @@
+use alloy_consensus::{Block, BlockHeader, Transaction, EMPTY_ROOT_HASH};
+use alloy_primitives::B256;
+use alloy_rpc_types_engine::{CancunPayloadFields, MaybeCancunPayloadFields};
+use derive_more::{Constructor, From, Into};
+
+/// Container type for all available additional `newPayload` request parameters that are not present
+/// in the [`ExecutionPayload`](alloy_rpc_types_engine::ExecutionPayload) object itself.
+///
+/// Default is equivalent to pre-cancun, payloads v1 and v2.
+#[derive(Debug, Clone, Default)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct OpExecutionPayloadSidecar {
+    /// Cancun request params introduced in `engine_newPayloadV3` that are not present in the
+    /// [`ExecutionPayload`](alloy_rpc_types_engine::ExecutionPayload).
+    cancun: MaybeCancunPayloadFields,
```

Comment:
should we add a note here that this is actually Canyon ? but equivalent to L1 cancun?